offchain/src/treasury/fund/index.ts

```diff
@@ -49,23 +51,37 @@ export async function fund<P extends Provider, W extends Wallet>({
   signers,
   metadata,
   vendor,
+  validFromSlot,
+  validUntilSlot,
 }: IFundArgs<P, W>): Promise<TxBuilder> {
   const { configs, scripts } = loadConfigsAndScripts(blaze, configsOrScripts);
   const registryInput = await blaze.provider.getUnspentOutputByNFT(
     AssetId(configs.treasury.registry_token + toHex(Buffer.from("REGISTRY"))),
   );
 
-  const maxHorizon = blaze.provider.network === NetworkId.Testnet ? 6 : 36;
-  const upperBoundUnix = Math.min(
-    Number(configs.treasury.expiration),
-    new Date().valueOf() + maxHorizon * 60 * 60 * 1000,
-  );
+  const tx = blaze.newTransaction().addReferenceInput(registryInput);
+
+  if (validFromSlot) {
+    tx.setValidFrom(Slot(validFromSlot));
+  } else {
+    tx.setValidFrom(Slot(blaze.provider.unixToSlot(Date.now())));
```

Comment:
Actually this wasn't being set in the old version so I don't actually know if we need it 🤔